### PR TITLE
Fix tb sfrd

### PR DESCRIPTION
**Pull request by EMventura**
Fixing 3 main bugs:

1) The velocity component in Tb assumed old (GADGET) grids. Fixing this to account for the SWIFT grids.
2) Fixing bug related to the supernova_feedback which was occurring when splitting the amount of m_eject gas.
3) Mistake in the Compton heating (we were not accounting for the Helium fraction f_He).

In addition to that, in a few spots a changed nomenclature and other minor things in order to get more similar to the version of the code that is used in Balu+22

### CODE REVIEW

**Comment by EMventura** [comment]
I haven't changed the NO_LIGHT statement as we saw it didn't matter at all, but if you prefer I can remove it. Before merging I am doing a final run as the test I did was on my scaling_relations branch while I modified this from the master branch.

**Comment by EMventura** [comment]
Final test done, if you are happy with the modifications we can merge

File: src/core/XRayHeatingFunctions.c
Comment (qyx268) [inline src/core/XRayHeatingFunctions.c:1443]: Brad was doing some debugging on the Ts calculation and noticed some difference between 21cmFAST and Meraxes. Despite all other differences between the two, this is another one that we didn't notice...

File: src/core/find_HII_bubbles.c
Comment (qyx268) [inline src/core/find_HII_bubbles.c:68]: this is neutral_fraction. am i misunderstanding something?

Comment (EMventura) [reply src/core/find_HII_bubbles.c:68]: I think it is neutral fraction but for some reason in Balu's version was called electron fraction so I changed the name for consistency... Do you want me to put it back?

File: src/core/BrightnessTemperature.c
Comment (EMventura) [reply src/core/BrightnessTemperature.c]: Just added. I also found that in read_grids.c there was not the case for the VELOCIRAPTOR_AUG so I added that one.
I was trying to test my modification but the code doesn't run when I put TreesID= 1 (GBPtrees) as I get this error:

"Failed to open file /fred/oz113/yqin/dragons/input_trees/Genesis_L210_N4320/trees/horizontal_trees_000.hdf5Task 0"

File: src/core/find_HII_bubbles.c
Comment (s-balu) [reply src/core/find_HII_bubbles.c:68]: mmmmm are you sure they are the same? If we are also taking into account He, they electron fraction is slightly more than the neutral fraction. Maybe this was named so for future development as well?

Comment (s-balu) [reply src/core/find_HII_bubbles.c:68]: https://github.com/s-balu/_meraxes_/blob/98d1d31b7ccda8bdf4230361dea86cced48ca57c/src/core/find_HII_bubbles.c#L412

It is set from the x_e value...

**Comment by EMventura** [comment]
I have tested also the version with the new commits, other than the error message when treesID = 1, everything else seems to work!

**Review by qyx268 — APPROVED** [review]

File: src/core/find_HII_bubbles.c
Comment (qyx268) [reply src/core/find_HII_bubbles.c:68]: @s-balu i think you misunderstood electron_fraction. electron_fraction is the fraction of electron, we are considering neutral hydrogen fraction. x_e is also the electron density within HI region.

File: src/core/BrightnessTemperature.c
Comment (qyx268) [reply src/core/BrightnessTemperature.c]: thanks for adding the VELOCIRAPTOR_AUG case for reading the grids!  I told Valentina to switch off 21cmfast in her project, which is the reason this didn't appear in her tests.

I think this failing is because you were still using `params/simulations/Genesis_L210_N4320.par` (?) so that `SimulationDir = /fred/oz113/yqin/dragons/input_trees/Genesis_L210_N4320`. To use Tiamat trees and set TreesID=1, we need to use e.g. Tiamat.par, so that `SimulationDir  =  /fred/oz013/simulations/Tiamat`. e.g.:

image

**Comment by qyx268** [comment]
happy to merge once we've settled down on electron_fraction vs neutral_fraction

File: src/core/find_HII_bubbles.c
Comment (s-balu) [reply src/core/find_HII_bubbles.c:68]: Well in that case it makes sense to revert (?) back to neutral_fraction.

Comment (EMventura) [reply src/core/find_HII_bubbles.c:68]: Ok so I will revert to neutral fraction and then happy to merge

**Comment by EMventura** [comment]
Merged and closing the pull request

**Pull request merged**
